File: study/ts/script/ui/component/CommonRechargeReward.ts
import { TypeConvertHelper } from "../../utils/TypeConvertHelper";
import { Lang } from "../../lang/Lang";
import { RichTextExtend } from "../../extends/RichTextExtend";
import UIHelper from "../../utils/UIHelper";

const {ccclass, property} = cc._decorator;

@ccclass
export default class CommonRechargeReward extends cc.Component {

    @property({
        type: cc.Node,
        visible: true
    })
    _panel_Content: cc.Node = null;

    @property({
        type: cc.Node,
        visible: true
    })
    _node_Text: cc.Node = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _image_Icon: cc.Sprite = null;


    updateUI(type, value, money, count) {
        var resParam = TypeConvertHelper.convert(type, value);
        var content = Lang.get('common_recharge_rewards', {
            money: money,
            count: count
        });
        var richText = RichTextExtend.createWithContent(content);
        richText.node.setAnchorPoint(cc.v2(0, 0));
        this._node_Text.addChild(richText.node);
        UIHelper.loadTexture(this._image_Icon, resParam.res_mini);
    }

}

File: study/lua/src/app/ui/component/CommonRechargeReward.lua
-- Author: panhao
-- Date:2018-11-23 17:11:36
-- Describle：

local CommonRechargeReward = class("CommonRechargeReward")

local EXPORTED_METHODS = {
    "updateUI",
}

function CommonRechargeReward:ctor()
    self._target = nil
    self._nodeDesc = nil
    self._imageIcon = nil
end

function CommonRechargeReward:_init()
    self._nodeDesc = ccui.Helper:seekNodeByName(self._target, "Node_Text")
    self._imageIcon = ccui.Helper:seekNodeByName(self._target, "Image_Icon")
end

function CommonRechargeReward:bind(target)
	self._target = target
	self:_init()
	cc.setmethods(target, self, EXPORTED_METHODS)
end

function CommonRechargeReward:unbind(target)
	cc.unsetmethods(target, EXPORTED_METHODS)
end

function CommonRechargeReward:updateUI(type, value, money, count)
    local TypeConvertHelper = require("app.utils.TypeConvertHelper")
	local resParam = TypeConvertHelper.convert(type, value)
	local content = Lang.get("common_recharge_rewards", {
			money = money,
			count = count,
		})
	local richText = ccui.RichText:createWithContent(content)
    richText:setAnchorPoint(cc.p(0, 0))
    self._nodeDesc:addChild(richText)
    self._imageIcon:loadTexture(resParam.res_mini)
end


return CommonRechargeReward